File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/java_to_py/code_output/Order.py
class Order:
    def __init__(self):
        self.menu = []
        self.selectedDishes = []
        self.sales = {}

    def addDish(self, dish):
        if dish is None or "dish" not in dish or "price" not in dish or "count" not in dish:
            return False

        for menuDish in self.menu:
            if dish["dish"] == menuDish["dish"]:
                dishCount = int(dish["count"])
                menuDishCount = int(menuDish["count"])

                if menuDishCount < dishCount:
                    return False
                else:
                    menuDish["count"] = menuDishCount - dishCount
                    self.selectedDishes.append(dish)
                    return True
        return False

    def calculateTotal(self):
        total = 0.0
        for dish in self.selectedDishes:
            dishName = str(dish["dish"])
            price = float(dish["price"])
            count = int(dish["count"])
            sale = self.sales.get(dishName, 1.0)
            total += price * count * sale
        return total

    def checkout(self):
        if not self.selectedDishes:
            return False
        total = self.calculateTotal()
        self.selectedDishes.clear()
        return total

File: data/derived_cot/rq3_prompting/results/translation/qwen/java_to_cpp/code_output/Order.cpp
#include <vector>
#include <unordered_map>
#include <string>
#include <any>
#include <optional>

class Order {
private:
    std::vector<std::unordered_map<std::string, std::any>> menu;
    std::vector<std::unordered_map<std::string, std::any>> selectedDishes;
    std::unordered_map<std::string, double> sales;

public:
    Order() = default;

    bool addDish(std::unordered_map<std::string, std::any> dish) {
        if (dish.empty() || !dish.contains("dish") || !dish.contains("price") || !dish.contains("count")) {
            return false;
        }

        for (auto& menuDish : menu) {
            if (std::any_cast<std::string>(dish["dish"]) == std::any_cast<std::string>(menuDish["dish"])) {
                int dishCount = std::any_cast<int>(dish["count"]);
                int menuDishCount = std::any_cast<int>(menuDish["count"]);

                if (menuDishCount < dishCount) {
                    return false;
                }

                menuDish["count"] = menuDishCount - dishCount;
                selectedDishes.push_back(dish);
                return true;
            }
        }
        return false;
    }

    double calculateTotal() const {
        double total = 0.0;
        for (const auto& dish : selectedDishes) {
            std::string dishName = std::any_cast<std::string>(dish.at("dish"));
            double price = std::any_cast<double>(dish.at("price"));
            int count = std::any_cast<int>(dish.at("count"));
            double sale = sales.at(dishName);

            total += price * count * sale;
        }
        return total;
    }

    std::any checkout() {
        if (selectedDishes.empty()) {
            return false;
        }

        double total = calculateTotal();
        selectedDishes.clear();
        return total;
    }
};